### ISSUE TITLE
Settings for purity

### ISSUE THREAD

**Issue by DanisAlukaev** [DanisAlukaev/XCBs#50]
I still think it may make sense to think of a problem where visual model alone will not be as strong. 80% purity, considering that some concepts may not be really active on this datasets, sounds pretty high to me. It may be too hard to improve on.   One option is to have objects which are somewhat visually different, but expressed in the same way in text and have the same influence on the classifier (i.e. the same 'semantics'). The vision conepts alone are likely to use different features for them, whereas KL should probably push it to have the same feature.  E.g., in a realistic setting, door handles can look pretty differently, but they fulfil the same function and we would refer to them in the same way (i.e. a 'door handle' and very rarely as 'octopus-shaped brass door handle'). 

It may also be a good idea because there is chance you will find hyperparameters which help the text-driven model to marginally improve on the vision-alone version. But then these hyperparameters may not generalize to the other datasets. I feel that getting the synthetic dataset which is better aligned with what we expect is a good idea (esp. since in this case we actually make it more realistic / challenging). 

And, as we descrined we should put some emphasis on the generalization setting, where we introduce some shortcuts (i.e. visual features which are predictive of the label in the training but not predictive on the generalization test set + not expressed in the text)

(But it is really good to see that generally learning seems to work quite well, actually)
